TwitchPlaysAssembly/Src/ComponentSolvers/Modded/TheCrazyCodr/SQLBasicComponentSolver.cs

```diff
@@ -0,0 +1,136 @@
+﻿using System;
+using System.Collections;
+using System.Threading;
+using UnityEngine;
+
+public class SQLBasicComponentSolver : ReflectionComponentSolver
+{
+	public SQLBasicComponentSolver(TwitchModule module) :
+		base(module, "SQLModule", "!{0} toggle [Toggles between showing the goal table and the editor] | !{0} SELECT X, Y, Z [Sets the SELECT clause] | !{0} WHERE W OPER X (AND/OR Y OPER Z) [Sets the WHERE clause, where the section in parentheses is optional] | !{0} LIMIT X, Y [Sets the LIMIT clause] | !{0} check/submit [Presses the check button]")
+	{
+		// Generated answer is not stored anywhere in the mod, best way I can think of to get it for autosolver
+		var source = DataComponentType.CallMethod<object>("FromDifficulty", null, _component.GetValue<object>("difficulty"));
+		var getAnswerThread = new Thread(() =>
+		{
+			targetQueryObj = Data2ComponentType.CallMethod<object>("GenerateFromDifficulty", null, _component.GetValue<object>("difficulty"), source);
+			while (_component.GetValue<object>("goal").ToString() != targetQueryObj.CallMethod<object>("Apply", source).ToString())
+				targetQueryObj = Data2ComponentType.CallMethod<object>("GenerateFromDifficulty", null, _component.GetValue<object>("difficulty"), source);
+			receivedAnswer = true;
+			return;
+		});
```

Comment:
This isn't necessarily a problem, but I'm not sure why this needs to be a Thread. Couldn't it just be a coroutine?

Reply:
I guess it could, I tried making it a Coroutine though and it didn't recognize StartCoroutine at all no matter what I did so I could not test this. However I notice that whenever Timwi has some algorithm which takes a while he usually puts it in a Thread, and I'm assuming its because putting it in a Coroutine would have the game or make it not as fast but I'm not sure.

Reply:
If you want to use StartCoroutine you need to call it on the module: `module.StartCoroutine`. Thinking about what you said, I guess it's probably better to be a Thread if this is doing a lot of calculations.